lib/sdcadm.js

```diff
@@ -1954,52 +2034,80 @@ SdcAdm.prototype.genUpdatePlan = function genUpdatePlan(options, cb) {
                     // All candidate images to `ch.images`. Just the single
                     // image if one was specified.
                     if (ch.image) {
-                        var imgName =
-                            self.config.imgNameFromSvcName[ch.service.name];
-
-                        self.getImage({uuid: ch.image}, function (iErr, img) {
-                            if (iErr) {
-                                nextChange(new errors.UpdateError(
-                                    iErr,
-                                    format('unknown image "%s" from %s',
-                                        ch.image, changeRepr)));
+                        self.imgNamesFromSvcName(ch.service.name,
+                                                 function (err, imgNames) {
+                            if (err) {
+                                nextChange(err);
                                 return;
                             }
-                            // Make sure we're not trying to update a service
-                            // to a different service image:
-                            if (img.name !== imgName) {
-                                nextChange(new errors.UpdateError(
-                                    format('invalid image "%s" for service ' +
-                                        '%s from %s. This image is for %s ' +
-                                        'service', ch.image, ch.service.name,
-                                        changeRepr, img.name)));
-                                return;
-                            }
-                            ch.images = [img];
-                            delete ch.image;
-                            nextChange();
+                            self.getImage({uuid: ch.image}, function (e, img) {
+                                if (e) {
+                                    nextChange(new errors.UpdateError(e,
+                                        format('unknown image "%s" from %s',
+                                            ch.image, changeRepr)));
+                                    return;
+                                }
+                                // Make sure we're not trying to update a
+                                // service to a different service image:
+                                if (imgNames.indexOf(img.name) === -1) {
```

Comment:
... because here, it looks like we're ok with any of the mantav1- or mantav2- service names, as defined up in defaults.json

Reply:
defaults.json has, e.g.:

```
"binder": ["mantav2-nameservice", "manta-nameservice"],
```

I.e. the "mantav1-" image names are *only for Manta services that are in a mantav1*.  Triton components will always use the "manta-" or "mantav2-" image names.